pinot-controller/src/main/resources/app/components/Homepage/InstancesTables.tsx

```diff
@@ -17,23 +17,30 @@
  * under the License.
  */
 
-import React, {  } from 'react';
-import map from 'lodash/map';
-import get from 'lodash/get';
+import React from 'react';
+import _ from 'lodash';
 import InstanceTable from './InstanceTable';
 
-const Instances = ({instances, clusterName}) => {
+const Instances = ({ instances, clusterName }) => {
   const order = ['Controller', 'Broker', 'Server', 'Minion'];
   return (
     <>
-      {
-        map(order, (key) => {
-          const value = get(instances, key, []);
-          return <InstanceTable key={key} name={`${key}s`} instances={value} clusterName={clusterName} />;
+      {_(order)
+        .filter((key) => _.has(instances, key))
+        .map((key) => {
+          const value = _.get(instances, key, []);
+          return (
+            <InstanceTable
+              key={key}
+              name={`${key}s`}
+              instances={value}
+              clusterName={clusterName}
+            />
+          );
```

Comment:
I think we don't need `.value` here, we are simply filtering instances that don't have key.

```js
{order
  .filter((key) => _.has(instances, key))
  .map((key) => {
    const value = _.get(instances, key, []);
    return (
      <InstanceTable
        key={key}
        name={`${key}s`}
        instances={value}
        clusterName={clusterName}
      />
    );
  })
}
```

Reply:
good call. forgot array already has those methods